internal/services/appservice/function_app_flex_consumption_resource.go

```diff
@@ -850,6 +886,14 @@ func (r FunctionAppFlexConsumptionResource) Update() sdk.ResourceFunc {
 				model.Properties.SiteConfig = siteConfig
 			}
 
+			arc, err := ExpandAlwaysReadyConfiguration(state.AlwaysReady, state.MaximumInstanceCount)
+			if err != nil {
+				return fmt.Errorf("expanding `always_ready` for %s: %+v", id, err)
+			}
+			if metadata.ResourceData.HasChange("always_ready") {
+				model.Properties.FunctionAppConfig.ScaleAndConcurrency.AlwaysReady = arc
+			}
+
```

Comment:
should we move this inside the `HasChange` statement since there is no need to expand this unless there's a change?

```suggestion
			if metadata.ResourceData.HasChange("always_ready") {
						arc, err := ExpandAlwaysReadyConfiguration(state.AlwaysReady, state.MaximumInstanceCount)
			if err != nil {
				return fmt.Errorf("expanding `always_ready` for %s: %+v", id, err)
			}
				model.Properties.FunctionAppConfig.ScaleAndConcurrency.AlwaysReady = arc
			}

```